k2/csrc/utils.cu

```diff
@@ -64,153 +82,180 @@ __global__ void RowSplitsToRowIdsKernel(int32_t num_rows,
   }
 }
 
-static int32_t RoundUpToNearestPowerOfTwo(int32_t n) {
-  K2_CHECK_GE(n, 0);
-  n--;
-  n |= n >> 1;
-  n |= n >> 2;
-  n |= n >> 4;
-  n |= n >> 8;
-  n |= n >> 16;
-  return n + 1;
+/*
+  This function works out the row_id of `this` index from row-splits, using
+  binary search.  Specifically, it returns i such that row_splits[i] <= index <
```

Comment:
mm, did you move this from bottom to top?  makes diff very hard to interpret.

Reply:
yeah, I lost the order. Anyway, would restore for the separated issue.